observer/baseNodeProvider_test.go

```diff
@@ -40,15 +42,42 @@ func TestBaseNodeProvider_InvalidNodesConfiguration(t *testing.T) {
 		},
 	}
 
-	bnp := baseNodeProvider{}
+	bnp := baseNodeProvider{
+		numOfShards: 1,
+	}
 	err := bnp.initNodes(nodes)
 	require.Contains(t, err.Error(), "observers for shard 1 must include at least one historical (non-snapshotless) observer")
 }
 
+func TestBaseNodeProvider_InvalidShardForObserver(t *testing.T) {
+	t.Parallel()
+
+	nodes := []*data.NodeData{
+		{
+			Address:        "addr0",
+			ShardId:        0,
+			IsSnapshotless: false,
+		},
+		{
+			Address:        "addr1",
+			ShardId:        2,
```

Comment:
shardId: 1 should also fail here

Reply:
true, updated, this way we test the mandatory greater or equal condition

Reply:
ok